diff -Nur '--exclude=.git' '--exclude=tuxtxt' enigma2-nightly.org/main/Makefile.am enigma2-nightly.patched/main/Makefile.am
--- enigma2-nightly.org/main/Makefile.am	2011-12-08 23:44:45.212475695 +0100
+++ enigma2-nightly.patched/main/Makefile.am	2011-12-08 19:23:24.000000000 +0100
@@ -58,8 +58,6 @@
 
 enigma2_LDFLAGS = -Wl,--export-dynamic
 
-enigma2_LDADD += -lglcddrivers -lglcdgraphics -lglcdskin
  
 if HAVE_GIT_DIR
 GIT_DIR = $(top_srcdir)/.git
 GIT = git --git-dir=$(GIT_DIR)
